plugins/out_stackdriver/stackdriver.c

```diff
@@ -1229,15 +1271,31 @@ static int stackdriver_format(struct flb_config *config,
         msgpack_pack_str_body(&mp_pck, "timestamp", 9);
 
         /* Format the time */
-        gmtime_r(&tms.tm.tv_sec, &tm);
-        s = strftime(time_formatted, sizeof(time_formatted) - 1,
-                     FLB_STD_TIME_FMT, &tm);
-        len = snprintf(time_formatted + s, sizeof(time_formatted) - 1 - s,
-                       ".%09" PRIu64 "Z", (uint64_t) tms.tm.tv_nsec);
-        s += len;
-
-        msgpack_pack_str(&mp_pck, s);
-        msgpack_pack_str_body(&mp_pck, time_formatted, s);
+        /* If format is timestamp_object or timestamp_duo_fields, 
+         * tms has been updated. 
+         * 
+         * If timestamp is not present or format is invalid_format_time,
+         * use the default tms(current time).
+         * 
+         * If format is time, directly use the flb_sds_t format_time.
+         */
```

Comment:
We should call `extract_timestamp` first to get `tms_status`, since `tms_status` is a parameter of `pack_json_payload`. It is used to remove time-related fields in the jsonPayload.